src/VisualStudio/Core/Def/Implementation/CodeLens/CodeLensCallbackListener.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.VisualStudio.Language.CodeLens;
+using Microsoft.VisualStudio.Language.CodeLens.Remoting;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Settings;
+using Microsoft.VisualStudio.Utilities;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.LanguageServices.CodeLens
+{
+    /// <summary>
+    /// This is used by new codelens API to get extra data from VS if it is needed.
+    /// </summary>
+    [Export(typeof(ICodeLensCallbackListener))]
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [ContentType(ContentTypeNames.VisualBasicContentType)]
```

Comment:
please file an editor bug to make it lazy, seems like it's going to be always loaded, I don't see it using any metadata

Reply:
@olegtk thank you for the review. can you tell me where I should open bug? for this particular issue, I think I l already told them about it. but let me file it again. and I found another issue.

Reply:
is there anything that is blocking this PR, @heejaechang, @jinujoseph ? Because you cannot disable References indicator, this is blocking our efforts to move legacy CodeLens provider to background thread.